### ISSUE TITLE
Bilinear Interpolate support?

### ISSUE THREAD

**Issue by Godricly** [microsoft/onnxruntime#2065]
**Describe the bug**
Model exported by pytorch can't inference.
**Urgency**
If there are particular important use cases blocked by this or strict project-related timelines, please share more information and dates. If there are no hard deadlines, please specify none.

**System information**
- OS Platform and Distribution (e.g., Linux Ubuntu 16.04): Ubuntu 18.04
- ONNX Runtime installed from (source or binary): source
- ONNX Runtime version: af8fe0f98004635e105fa5031f1c3de8381e86b4
- Python version:3.6
- Visual Studio version (if applicable):
- GCC/Compiler version (if compiling from source):
- CUDA/cuDNN version: 10.1 cudnnv7
- GPU model and memory: 1080 8G

**To Reproduce**
```
import torch
import onnxruntime
from torch import nn
from torch.nn import functional as F
import numpy as np

class Demo(torch.nn.Module):
    def __init__(self):
        super().__init__()
        self.conv1 = nn.Conv2d(16,16,3,2,1)

    def forward(self, x):
        y = self.conv1(x)
        y = F.relu(y)
        y_up = F.interpolate(y, x.shape[-2:], mode='bilinear', align_corners=False)
        return y_up + x

if __name__ == "__main__":
    input_tensor = torch.zeros((1,16,100,100))
    demo = Demo()
    out = demo(input_tensor)
    torch.onnx.export(demo, input_tensor, "debug.onnx", verbose=True,
                        input_names=['data'],
                        opset_version=11,
                        # do_constant_folding=True,
                        dynamic_axes={'data':{0:'batch', 2:'width', 3:'height'},
                        })

    ses_option = onnxruntime.SessionOptions()
    ort_session = onnxruntime.InferenceSession("debug.onnx")#, ses_option)


    dummy = np.zeros((1,16,100,100), dtype=np.float32)
    ort_inputs = {ort_session.get_inputs()[0].name: dummy}
    ort_outs = ort_session.run(None, ort_inputs)
```

**Expected behavior**
```
onnxruntime.capi.onnxruntime_pybind11_state.NotImplemented: [ONNXRuntimeError] : 9 : NOT_IMPLEMENTED : Could not find an implementation for the node Resize(11)
```

**Screenshots**

**Additional context**
Pytorch INFO
```
Collecting environment information...
PyTorch version: 1.3.0.dev20191007
Is debug build: No
CUDA used to build PyTorch: 10.0

OS: Ubuntu 18.04.3 LTS
GCC version: (Ubuntu 7.4.0-1ubuntu1~18.04.1) 7.4.0
CMake version: version 3.15.3

Python version: 3.6
Is CUDA available: Yes
CUDA runtime version: 10.1.243
GPU models and configuration: GPU 0: GeForce GTX 1080
Nvidia driver version: 430.50
cuDNN version: /usr/lib/x86_64-linux-gnu/libcudnn.so.7.5.0

Versions of relevant libraries:
[pip3] numpy==1.17.2
[pip3] torch==1.3.0.dev20191007
[pip3] torchvision==0.2.2.post3
[conda] Could not collect
```

**Comment by hariharans29** [comment]
You have an opset-11 model. OnnxRuntime does not fully support opset-11 yet. It will only support opset-11 in its next release. Opset-10 models should run. Please try converting to opset-10 or wait for the next release in which opset-11 will be supported.

**Comment by hariharans29** [comment]
Actually keeping this opened for tracking.

CC: @askhade

**Comment by snippler** [comment]
Could you explain what you mean by your reply? The problem seems to be that mode='bilinear' is not supported. Changing opset_version does not seem to change this. If instead I change to mode='nearest' then it works.

**Comment by hariharans29** [comment]
Hello,

Your model was an opset-11 model and we had not completed the implementation for opset-11 Resize at that point. It is now supported via #2088 . Could you please re-build from master and give it a try ? Thanks.

**Comment by Godricly** [comment]
```
/onnxruntime/onnxruntime/core/providers/cpu/tensor/upsample.h:228 void onnxruntime::UpsampleBase::ParseScalesData(const onnxruntime::Tensor*, std::vector<float>&) const scales_size > 0 was false. scales size should be greater than 0.
```
Its not working yet. U can try with demo code above. I'm not sure it is a bug in torch or onnxruntime. The scaling size is is not right.

**Comment by askhade** [comment]
@Godricly : There was a fix checked in for this yesterday night... can you try again...
Pytorch converters add empty scales as constant initializer and use sizes to set the output dimensions... there was a check missing for this which has now been checked in....

**Comment by Godricly** [comment]
```
2019-10-23 15:10:48.572430561 [E:onnxruntime:, sequential_executor.cc:165 Execute] Non-zero status code returned while running Resize node. Name:'' Status Message: onnxruntime/onnxruntime/core/providers/cpu/tensor/upsample.h:221 void onnxruntime::UpsampleBase::ScalesValidation(const std::vector<float>&, onnxruntime::UpsampleMode) const scales.size() == 2 || (scales.size() == 4 && scales[0] == 1 && scales[1] == 1) was false. 'Linear' mode only support 2-D inputs ('Bilinear') or 4-D inputs with the corresponding outermost 2 scale values being 1 in the Resize operator
```
The session creation is passed.  But the execution is not working.
@askhade

**Comment by askhade** [comment]
What is your scales shape?

**Comment by Godricly** [comment]
Is there any update for this one?

**Comment by askhade** [comment]
I dont quite understand how you are constructing your onnx resize node from this :
ort_inputs = {ort_session.get_inputs()[0].name: dummy}
ort_outs = ort_session.run(None, ort_inputs)

To make this faster can you clarify what is the shape of your input, and what exactly is your scale?

For Linear mode we only support BiLinear meaning scale can be either [x, y] or [1, 1, x, y] 
This is the exact check which is failing so I suspect your scale does not conform to this...

**Comment by Godricly** [comment]
I have no idea what you are asking for. Basically, I export the ONNX model of pytorch with bilinear mode interpolation. And those 2 lines has nothing to do with resize node at all. The input could be in any shape(100X100 is used here for demonstration). In the demo code, the input is downsampled with a stride of 2(50X50 now), which also is the input for the resize OP.  The expected output for resize is 100X100.

**Comment by Godricly** [comment]
[#27176](https://github.com/pytorch/pytorch/issues/27176) could be explanation.

**Comment by Godricly** [comment]
problem fixed with night build pytorch(1.4.0.dev20191103).

**Comment by glenn-jocher** [comment]
I can not export from PyTorch to ONNX using the upsample operator with torch version 1.3.1, 
onnx version 1.5.0.

MINIMUM CODE TO REPRODUCE:
```python
import torch
import torch.nn as nn
import torch.nn.functional as F
import onnx

print(torch.__version__)
print(onnx.__version__)


class TestModel(nn.Module):
    def __init__(self):
        super(TestModel, self).__init__()

    def forward(self, x):
        x = F.interpolate(x, scale_factor=2, mode='nearest')
        return x


torch_model = TestModel()
dummy_input = torch.randn(1, 3, 256, 256)

torch_out = torch.onnx.export(torch_model, dummy_input, 'model.onnx', verbose=True, opset_version=11)

onnx_model = onnx.load('model.onnx')
print(onnx_model)
onnx.checker.check_model(onnx_model)
```

Produces the following error:
```bash
Traceback (most recent call last):
  File "/Users/glennjocher/PycharmProjects/iD/upsample_error_reproduce.py", line 26, in <module>
    onnx.checker.check_model(onnx_model)
  File "/Users/glennjocher/.conda/envs/yolov3/lib/python3.7/site-packages/onnx/checker.py", line 86, in check_model
    C.check_model(model.SerializeToString())
onnx.onnx_cpp2py_export.checker.ValidationError: Node () has input size 4 not in range [min=2, max=2].

==> Context: Bad node spec: input: "input" input: "23" input: "23" input: "22" output: "24" op_type: "Resize" attribute { name: "coordinate_transformation_mode" s: "asymmetric" type: STRING } attribute { name: "cubic_coeff_a" f: -0.75 type: FLOAT } attribute { name: "mode" s: "nearest" type: STRING } attribute { name: "nearest_mode" s: "floor" type: STRING }
```
